### PR TITLE
Space Encounters Expansion

### PR DESCRIPTION
**Pull request by Valo56**
This was a fun one to put together.
-Neutral NPC freighters that players can attack and destroy. These are more powerful than the pirates normally encountered, but offer much better loot!
-Aurek Strikefighters and Sith Fighters. Sith Fighters also spawn at a high rate in Korriban space.
-Hostile NPC capital ships that players will need to group up to take on, requiring a high degree of coordination (tanking, repair and DPS ships) in order to overcome, but with rewards in line with their difficulty.
-New NPC-only weapons for both of these.
-Tweaked spawns across the board in space. The distribution of NPC's won't be much different than it was before, the numbers just needed to change in order to accomodate the uncommon/rare freighter spawns and the super rare capital ship spawns (the highest tier capital ship, which players are unlikely to be able to beat even if they all group up, only has a roughly 1 in 50-60 thousand chance of spawning)

As a side note I have also adjusted the Republic Base Exterior and the Sith Lake Base to use new NPC's that work with revamp. New non-hostile Sith NPC's are available in the NPC's > Elves section, with no difference from their hostile counterparts in the Dwarves section.

I tried to make it so armor and shield repairers could be used on self for this PR but I wasn't able to get it to work. :( That'll have to be someone else unless I can make some breakthrough on it another time.

### CODE REVIEW

File: SWLOR.Game.Server/Core/NWScript/Enum/Appearance.cs [hunk 1]
```diff
@@ -411,6 +411,7 @@ public enum AppearanceType
         // SWLOR Starts Here
 
         TiefighterSM = 870,
+        LargeShuttle = 892,
```
Comment (zunath) [inline SWLOR.Game.Server/Core/NWScript/Enum/Appearance.cs:414]: Just confirming: Do these appearances already exist in our haks? Were we just missing these enumerations?

File: SWLOR.Game.Server/Feature/ShipModuleDefinition/CombatLaserModuleDefinition.cs [hunk 2]
```diff
@@ -21,18 +21,30 @@ public class CombatLaserModuleDefinition: IShipModuleListDefinition
             CombatLaser("com_laser_2", "Combat Laser II", "Cmbt Laser II", "Deals 17 thermal DMG to your target.", 3, 5f, 12, 17);
             CombatLaser("com_laser_3", "Combat Laser III", "Cmbt Laser III", "Deals 21 thermal DMG to your target.", 4, 6f, 15, 21);
             CombatLaser("com_laser_4", "Combat Laser IV", "Cmbt Laser IV", "Deals 26 thermal DMG to your target.", 5, 7f, 18, 26);
+            CombatLaser("cap_pdl_1", "Capital Point Defense I", "Cap PDL I", "A corvette's fast firing point defense. 15 DMG.", 5, 2f, 2, 15);
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/ShipModuleDefinition/CombatLaserModuleDefinition.cs:24]: I think these are NPC items but let's go ahead and keep these focused on functionality and exclude any flavor text. Can you change the descriptions to be in line with the existing laser descriptions? E.G:

"Deals 15 thermal DMG to your target."

File: SWLOR.Game.Server/Core/NWScript/Enum/Appearance.cs (hunk 1 above)
Comment (Valo56) [reply SWLOR.Game.Server/Core/NWScript/Enum/Appearance.cs:414]: Yes, they all exist in appearance.2da, I use them sometimes for events. I've tested the server and everything seems to spawn and function appropriately.

File: SWLOR.Game.Server/Feature/ShipModuleDefinition/IonCannonModuleDefinition.cs [hunk 3]
```diff
@@ -21,18 +21,25 @@ public class IonCannonModuleDefinition : IShipModuleListDefinition
             IonCannon("ion_cann_2", "Ion Cannon II", "Ion Cann. II", "Deals 17 EM DMG to your target.", 3, 5f, 12, 17);
             IonCannon("ion_cann_3", "Ion Cannon III", "Ion Cann. III", "Deals 21 EM DMG to your target.", 4, 6f, 15, 21);
             IonCannon("ion_cann_4", "Ion Cannon IV", "Ion Cann. IV", "Deals 26 EM DMG to your target.", 5, 7f, 18, 26);
+            IonCannon("cap_turbo_1", "Capital Ship Turbolaser I", "Cap Turbo I", "A corvette's main guns. 35 DMG.", 5, 15f, 20, 45);
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/ShipModuleDefinition/IonCannonModuleDefinition.cs:24]: There are a lot of duplicates here. If the stats are exactly the same, there's no reason to introduce two of the same type of item. Please remove the duplicates and rename the items accordingly.

File: SWLOR.Game.Server/Feature/ShipModuleDefinition/MissileLauncherModuleDefinition.cs [hunk 4]
```diff
@@ -21,6 +21,13 @@ public class MissileLauncherModuleDefinition: IShipModuleListDefinition
             MissileLauncher("msl_launch_2", "Missile Launcher II", "Msl Launch II", "Deals 15 explosive DMG to your target and nearby ships.", 3, 14f, 18, 15);
             MissileLauncher("msl_launch_3", "Missile Launcher III", "Msl Launch III", "Deals 19 explosive DMG  to your target and nearby ships.", 4, 17f, 22, 19);
             MissileLauncher("msl_launch_4", "Missile Launcher IV", "Msl Launch IV", "Deals 24 explosive DMG to your target and nearby ships.", 5, 20f, 26, 24);
+            MissileLauncher("cap_missile_1", "Concussion Missile I", "CapMissile1", "A corvette's concussion missile launchers.", 5, 18f, 30, 25);
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/ShipModuleDefinition/MissileLauncherModuleDefinition.cs:24]: Duplicates here too.

**Review by zunath — CHANGES_REQUESTED** [review]

File: SWLOR.Game.Server/Feature/ShipModuleDefinition/IonCannonModuleDefinition.cs (hunk 3 above)
Comment (Valo56) [reply SWLOR.Game.Server/Feature/ShipModuleDefinition/IonCannonModuleDefinition.cs:24]: These are less duplicates and more different recast. For instance, firing once every 2.5 rounds (15f) vs every 2 rounds (12f). It's a 20% DPR change. I felt that difference in recast time was better than just making bigger ships fire bigger guns every time. That said I might change it to 15f and 9f to differentiate them better.

File: SWLOR.Game.Server/Feature/ShipModuleDefinition/MissileLauncherModuleDefinition.cs (hunk 4 above)
Comment (Valo56) [reply SWLOR.Game.Server/Feature/ShipModuleDefinition/MissileLauncherModuleDefinition.cs:24]: Yeah these ones are bad gonna redo them.

**Pull request merged**